py3/chapter15/open_imap.py

```diff
@@ -22,10 +22,7 @@ def main():
         print('Listing mailboxes:')
         data = c.list_folders()
         for flags, delimiter, folder_name in data:
-            print('  %-30s%s %s' %
-                    ( ( b' '.join(flags) ).decode() ,
-                      delimiter.decode() ,
-                      folder_name ) )
+            print('  %-30s%s %s' % ( b' '.join(flags), delimiter, folder_name ) )
```

Comment:
Nice! I like how your syntax is moving more towards PEP-8. There's just a little extraneous whitespace to clean up and you'll have arrived. (And then you'll just be waiting on me to figure out how to get a test environment running again, and who knows how long that might take me. Alas!)

Reply:
Thanks for the encouragement; first time I've actually submitted a pull request for something like this.

I'll pull out the whitespace and push again.

Your book is very helpful, and I'm going to promote it in the "Python and IMAP" article I'm working on.